src/Factories/ComponentFactory.ts

```diff
@@ -89,14 +89,19 @@ export namespace ComponentFactory {
         return 'public';
     }
 
-    export function getHeritageClauseNames(heritageClause: ts.HeritageClause, checker: ts.TypeChecker): string[] {
+    export function getHeritageClauseNames(heritageClause: ts.HeritageClause, checker: ts.TypeChecker): string[][] {
         return heritageClause.types.map((nodeObject: ts.ExpressionWithTypeArguments) => {
             const symbolAtLocation: ts.Symbol | undefined = checker.getSymbolAtLocation(nodeObject.expression);
             if (symbolAtLocation !== undefined) {
-                return checker.getFullyQualifiedName(symbolAtLocation);
+                // tslint:disable
+                let ogSymbol: any = checker.getAliasedSymbol(symbolAtLocation);
```

Comment:
Can the type be used here versus `any`? Using `any` will make this difficult to spot issues at compile time.

Reply:
Thanks...I also said in the original comment I don't know what the solution to this typing issue is here and if you give me the answer I can include it.

Reply:
@bafolts 
I think I found the issue with this, should be able to adjust momentarily.

Reply:
@bafolts
Everything is looking good. Please excuse the bitwise, this one looks ignorable :)